src/patterns/kinesis-lambda.ts

```diff
@@ -35,13 +54,24 @@ import type { StreamErrorHandlingProps, StreamProcessingProps } from "../utils/l
  * existingKinesisStream: { existingLogicalId: "MyCloudFormedKinesisStream" }
  * ```
  *
- * Alternatively, reference a Kinesis stream which belongs to another stack or pattern using:
+ * Alternatively, reference a Kinesis stream which belongs to another stack or pattern in the same aws account using:
  * ```typescript
  * existingKinesisStream: { externalKinesisStreamName: "KinesisStreamFromAnotherStack" }
  * ```
+ *
+ *  Alternatively, reference a Kinesis stream which belongs to another stack or pattern in a different aws account using:
+ * ```typescript
+ * existingKinesisStream: {
+ *   crossAccountKinesisStream: {
+ *      roleArn: "CrossAccountKinesisStreamRoleArn",
+ *      streamArn: "CrossAccountKinesisStreamArn",
+ *   }
+ * }
+ * ```
  */
 export interface ExistingKinesisStream extends GuMigratingResource {
   externalKinesisStreamName?: string;
+  crossAccountKinesisStream?: CrossAccountKinesisStream;
```

Comment:
Interesting point 🤔  So this way, the consumer of the `ExistingKinesisStream` would always need to give the full `arn` rather than the stream name?

And then, if the stream is on a different account, the `roleArn` will also be populated, otherwise, the stream is on the same account?

Reply:
@akash1810 I made the changes in a separate branch that's based on the current branch of this PR. Could you have a look please? https://github.com/guardian/cdk/pull/1265